translation: Onboarding.md (#24)
opensource-way-chinese-translation/the_open_source_way-guidebook-2.0-zh: translation/Growing Contributors/onboarding.md

### 入职

**介绍**

贡献者是任何开源项目的生命线。一个项目如果不去不断地欢迎新的参与者，可能会发现自己停滞不前，或者更糟糕———项目会分崩离析。

因此，任何希望长期持续发展的开源项目都应该认真考虑它与新贡献者的联系、迎接和适应的方式。一句话，他们是如何 "上岗"的。一个精心设计的入职经历会使参与者更容易为项目做出贡献，并使贡献者能够尽快对工作产生积极影响。

在这一章中，我们将讨论你的社区的入职经历的重要性，并研究一些常见的入职资源和做法，这可能有助于你的项目变得更加可持续。

**入职经历的重要性**

> 在许多方面，在一个新的开源社区中的入职经历与你在一份新工作中的体验非常相似。

Reviewer: ```suggestion
在许多方面，一个新的开源社区中的入职经历与你在一份新工作中的体验非常相似。
```

例如，当你开始在一个新的职位（或在一个全新的组织）工作时，你需要熟悉新的人、新的工具、新的流程、新的规范、新的政策，等等。而你很可能会觉得自己有动力快速完成这些事情，因为你渴望向组织展示你的价值，并在新环境中变得舒适，这样你就可以在最佳状况做你的工作。你越快感觉到自己正在产生影响，你就越有可能留在这里。而你也会越发满意。

> 但是，你也可能加入一个有着广泛历史的团队--存在共同的规范和习俗、内部笑话、行话、不言而喻的协议、共同的纽带和集体认同感。简而言之，你的新团队或组织会有一种在你加入之前就存在的文化。这种文化的各个方面在很大程度上是无形的，但它们会大大影响你与他人合作的能力。一个有效的入职体验能让你了解这种文化。

Reviewer: ```suggestion
但是，你也可能加入一个有着广泛历史的团队——存在共同的规范和习俗、内部笑话、行话、不言而喻的协议、共同的纽带和集体认同感。简而言之，你的新团队或组织会有一种在你加入之前就存在的文化。这种文化的各个方面在很大程度上是无形的，但它们会大大影响你与他人合作的能力。一个有效的入职体验能让你了解这种文化。
```

中文标点

这些对于开源社区的入职体验来说都是很真实的。在入职方面的挑战包括：

**异步远程环境**

> ```

Reviewer: @Cubik65536 能帮助确认一下这里是代码符号吗？我认为这里可能是引用或者其他的。Thx

Reviewer 2: @KevinZonda 没有明白你想问的事情，原文这里确实是粗体字

传统办公环境的一大好处是，你可以快速找到成员并提问，与同事和经理进行面对面的交流。相反，当你加入一个新的开源社区时，你能会与不同地理位置和时区的社区成员一起工作。因此，除非有好的入门材料，如文档、视频教程、wiki页面和其他资源，否则即使对于有经验的主动志愿者来说，在开源社区起步也将更具挑战性。
```

**冒充者综合症**

```
当获得实时帮助或反馈比较困难时，人们很容易被简单的工具问题困住，且对自己的工作感到不自信。因此，在开源社区中，良好的入职体验对于新人在社区的早期不感到气馁是至关重要的。
```

**显著的多元性**

> ```

Reviewer: 同上 @Cubik65536

没有哪两个开源社区是相同的。即使是经验丰富的开源贡献者，当他们加入一个新的社区时，也需要熟悉新的工具、新的流程，甚至是新的术语。根据不同的社区，像贡献、社区成员、项目、上游等这些看似简单的术语，其含义都可能略有不同，很容易给新的社区成员带来困惑。
```

**入职基础**

> 当你努力使你的项目的入职经历和体验最有效时，问问自己这些问题：

Reviewer: Sigh...

- 人们能为项目做出什么贡献？
- 人们在哪里为项目做出贡献？
- 人们如何为项目做出贡献？

回答关于你的项目入职经历的第一个问题需要做大量的工作。新来的贡献者需要知道什么是他们可以做出的最有用的贡献。相当多的项目在他们的源代码库中设置了 "首次贡献者" 或类似的标签，这是引导新来的贡献者去解决问题和错误的一个很好的方法，同时也向新人介绍了项目的原始资料。

> 第二个问题--关于 "在哪里贡献"————这是一个并不像你想象的那样在项目里会经常提出的问题。一些项目老手可能会不屑一顾地说，"你需要的东西在论坛上" 或 "如果你需要更多的细节，请去阅读wiki"。但是，在哪个具体的资源库里有这些材料？在什么组织下？还有其他类似的项目呢？确保你的项目网站有明确的链接，让新的贡献者能够对项目产生影响。这些链接也不应该被埋没；如果可能的话，它们应该就在主页上。

Reviewer: ```suggestion
第二个问题——关于 "在哪里贡献"——这是一个并不像你想象的那样在项目里会经常提出的问题。一些项目老手可能会不屑一顾地说，"你需要的东西在论坛上" 或 "如果你需要更多的细节，请去阅读wiki"。但是，在哪个具体的资源库里有这些材料？在什么组织下？还有其他类似的项目呢？确保你的项目网站有明确的链接，让新的贡献者能够对项目产生影响。这些链接也不应该被埋没；如果可能的话，它们应该就在主页上。
```

你确定不修正一下符号。PS，这个suggestion不包含对引号的修正。

### 入职培训

**介绍**

贡献者是任何开源项目的生命线。一个项目如果不去不断地欢迎新的参与者，可能会发现自己停滞不前，或者更糟糕———项目会分崩离析。

> 因此，任何希望长期持续发展的开源项目都应该认真考虑它与新贡献者的联系、迎接和适应的方式。一句话，他们是如何 "上岗"的。一个精心设计的入职过程会使参与者更容易为项目做出贡献，并使贡献者能够尽快对工作产生积极影响。

Reviewer 3: ```suggestion
因此，任何希望长期持续发展的开源项目都应该认真考虑它与新贡献者的联系、迎接和适应的方式。一句话，他们是如何 "上岗"的。一个精心设计的入职经历会使参与者更容易为项目做出贡献，并使贡献者能够尽快对工作产生积极影响。
```

> 在这一章中，我们将讨论你的社区的入职过程的重要性，并研究一些常见的入职资源和做法，这可能有助于你的项目变得更加可持续。

Reviewer 3: ```suggestion
在这一章中，我们将讨论你的社区的入职经历的重要性，并研究一些常见的入职资源和做法，这可能有助于你的项目变得更加可持续。
```

**入职体验的重要性**

在许多方面，在一个新的开源社区中的入职体验与你在一份新工作中的体验非常相似。

例如，当你开始在一个新的职位（或在一个全新的组织）工作时，你需要熟悉新的人、新的工具、新的流程、新的规范、新的政策，等等。而你很可能会觉得自己有动力快速完成这些事情，因为你渴望向组织展示你的价值，并在新环境中变得舒适，这样你就可以在最佳状况做你的工作。你越快感觉到自己正在产生影响，你就越有可能留在这里。而你也会越发满意。

但是，你也可能加入一个有着广泛历史的团队--存在共同的规范和习俗、内部笑话、行话、不言而喻的协议、共同的纽带和集体认同感。简而言之，你的新团队或组织会有一种在你加入之前就存在的文化。这种文化的各个方面在很大程度上是无形的，但它们会大大影响你与他人合作的能力。一个有效的入职体验能让你了解这种文化。

这些对于开源社区的入职体验来说都是很真实的。在入职方面的挑战包括：

**异步远程环境**

```
传统办公环境的一大好处是，你可以快速找到成员并提问，与同事和经理进行面对面的交流。相反，当你加入一个新的开源社区时，你能会与不同地理位置和时区的社区成员一起工作。因此，除非有好的入门材料，如文档、视频教程、wiki页面和其他资源，否则即使对于有经验的主动志愿者来说，在开源社区起步也将更具挑战性。
```

**冒充者综合症**

```
当获得实时帮助或反馈比较困难时，人们很容易被简单的工具问题困住，且对自己的工作感到不自信。因此，在开源社区中，良好的入职体验对于新人在社区的早期不感到气馁是至关重要的。
```

**巨大的可变性**

```
没有哪两个开源社区是相同的。即使是经验丰富的开源贡献者，当他们加入一个新的社区时，也需要熟悉新的工具、新的流程，甚至是新的术语。根据不同的社区，像贡献、社区成员、项目、上游等这些看似简单的术语，其含义都可能略有不同，很容易给新的社区成员带来困惑。
```

**入职基础**

> 当你努力使你的项目的入职过程和体验最有效时，问问自己这些问题：

Reviewer 3: ```suggestion
当你努力使你的项目的入职经历最有效果时，问问自己这些问题：
```

- 人们能为项目做出什么贡献？
- 人们在哪里为项目做出贡献？
- 人们如何为项目做出贡献？

> 回答关于你的项目入职过程的第一个问题需要做大量的工作。新来的贡献者需要知道什么是他们可以做出的最有用的贡献。相当多的项目在他们的源代码库中设置了 "首次贡献者" 或类似的标签，这是引导新来的贡献者去解决问题和错误的一个很好的方法，同时也向新人介绍了项目的原始资料。

Reviewer 3: ```suggestion
回答关于你的项目入职经历的第一个问题需要做大量的工作。新来的贡献者需要知道什么是他们可以做出的最有用的贡献。相当多的项目在他们的源代码库中设置了 "首次贡献者" 或类似的标签，这是引导新来的贡献者去解决问题和错误的一个很好的方法，同时也向新人介绍了项目的原始资料。
```

第二个问题--关于 "在哪里贡献"————这是一个并不像你想象的那样在项目里会经常提出的问题。一些项目老手可能会不屑一顾地说，"你需要的东西在论坛上" 或 "如果你需要更多的细节，请去阅读wiki"。但是，在哪个具体的资源库里有这些材料？在什么组织下？还有其他类似的项目呢？确保你的项目网站有明确的链接，让新的贡献者能够对项目产生影响。这些链接也不应该被埋没；如果可能的话，它们应该就在主页上。

> 第三个问题是“如何贡献”的问题。有的项目经常公布源代码，来使其易于发现。。。对的，就只是这样。你可能听说过其他人把这种方法称为“把它扔到墙上（throwing it over the wall）”，但这不是一个人们倾向于礼貌地使用的词语。太多的项目会将其代码或内容直接推到公共场所，宣布对所有人开放，然后想知道所有可能的贡献者都在哪里。这可能是因为在项目中缺乏获得贡献的文件和程序，且存在真正的加入障碍。

Reviewer 3: ```suggestion
第三个问题是“如何贡献”的问题。有的项目经常公布源代码，让人能容易发现这个项目。对，就是如此。你可能听说过其他人把这种方法称为“把它扔到墙上（throwing it over the wall）”，但这不是一个人们倾向于礼貌地使用的词语。太多的项目会将其代码或内容直接推到公共场所，宣布对所有人开放，然后想知道所有可能的贡献者都在哪里。这可能是因为在项目中缺乏获得贡献的文件和程序，且存在真正的加入障碍。
```

接下来，让我们讨论一下你的项目可以使用的一些资源和实践，以消除这些障碍并简化项目的入职流程。

**入职资源**

在异步环境中成功的工作（和实践）的一个关键工具是良好的文档。由于贡献者通常不会有社区成员在他们附近提供即时指导，所以重要的是要把重要的社区规范、决定和流程记录下来，而且要比在更传统的工作环境中更详细。无论是登陆页面、项目文档、wiki页面等，一套好的参考资料对于新的社区成员熟悉社区并开始工作至关重要。确保入职文件是最新的，反映社区活动的最新快照也很重要，所以你要让任何人都能轻松地帮助保持内容的更新。一个易于使用的文档工具肯定有助于此，但更重要的是培养一种社区文化，让每个人都明白需要可靠、更新的文档。

这可能听起来很显而易见————入职培训的另一个关键资源是其他社区成员。即使人们不在同一个办公室，但能够接触到有经验的社区成员，帮助人们开始工作并回答问题，也可以帮助缓解早期的孤独感。对于一个相对较新的小型社区来说，一些简单的方式，比如可以在欢迎消息里跳出来的入职伙伴可能就足够了。对于更大和更成熟的社区，你会见到专注于入职培训的工作小组，甚至是有正式头衔的社区成员，比如教练或导师。

一旦人们的入职资源到位，这些信息（尤其是关于向谁寻求帮助的信息）需要在多个地方发布，这样人们就会因为与他们相关联的入职资源感到欢迎，并感到被鼓励去问问题。如果获得帮助很困难，新来的人就会对参与社区活动感到气馁，甚至认为这个社区可能不适合他们。

无论是入职伙伴、导师、工作小组等，你都应该希望有足够多的志愿者，这样这些志愿者就不会因为教导入职活动而感到疲惫，特别是要随着社区的发展，在社区内形成一种文化，使每个人都有帮助欢迎新社区成员的期望。理想情况下，我们希望看到的是很多人自愿帮助他人，无论他们是否有教练或导师的正式称号。如果你有一个正式的入职工作组、导师等项目，你不应该为社区成员成为正式的入职资源组人员设置高门槛。最重要的资格应该是人们帮助他人的意愿，而不是其他因素，像是他们在社区的任期或项目的技术专长。帮助入职的人不需要有所有的这些东西。相反，他们需要帮助新人更快的找到答案，而不是让新人独自工作。

**入职实践**

> 当人们加入一个新的社区时，他们可能会发现在交流渠道（如邮件或Slack）上的讨论让人不知所措。为了减轻他们的困惑或压力，可以考虑在这些频道中,为新的社区成员创造空间，让他们在开始和适应的时候使用。诸如 "入门 "或 "介绍 "这样的专用频道可以成为为人们提供安全感的地方，以便新成员提问。这些频道也允许更多有经验的社区成员通过帮助回答新用户的问题，甚至只是提供鼓励来协助入职。这对使新的社区成员在开始时感到受欢迎有很大的帮助。

Reviewer 3: ```suggestion
当人们加入一个新的社区时，他们可能会发现在交流渠道（如邮件或Slack）上的讨论让人不知所措。为了减轻他们的困惑或压力，可以考虑在这些频道中，为新的社区成员创造空间，让他们在开始和适应的时候使用。诸如 "入门 "或 "介绍 "这样的专用频道可以成为为人们提供安全感的地方，以便新成员提问。这些频道也允许更多有经验的社区成员通过帮助回答新用户的问题，甚至只是提供鼓励来协助入职。这对使新的社区成员在开始时感到受欢迎有很大的帮助。
```

一旦新的社区成员觉得自己有足够的能力开始为一个开源项目做出贡献，他们的下一个问题将是 "我在哪里可以做出贡献？"

因此，为了帮助新来的人开始工作，你应该制定一份工作清单，特别是新来的社区成员可以处理的工作项目。以及社区成员可以处理的工作项目。

许多社区在他们的issue（译者注：问题追踪器）中给问题贴上标签，如 "适合首次贡献者"、"好的第一个issue "或 "需要帮助"，这样新成员可以更容易地确定他们可以立即提供帮助的任务。标签的范围可以是文档错误、简单的错误修复，或其他简单的任务，这些任务将帮助新的贡献者体验早期的成功，从而建立他们的信心。

在这些问题上列出一个（或几个）作为导师或教练的联系人（以备人们需要帮助时）也是有帮助的。永远记住：对有经验的社区成员来说可能很简单的问题，对他们来说可能不那么简单。

许多开源社区组织活动旨在联系他们的成员。无论这些活动是在同一地点还是虚拟的，它们都为社区成员提供了同步合作的绝佳机会，并在这个过程中相互了解。这些活动可以是峰会、黑客马拉松、用户会议等。在这些活动中，可以考虑为新的社区成员创造特殊的程序或空间。

> 如果你的预算允许，你可以组织正式的迎新会，作为一个 "day 0" 活动，或者在午餐时间组织迎新会，让新成员认识其他社区成员，这样他们就知道以后可以向谁提问了。

Reviewer 3: ```suggestion
如果你的预算允许，你可以组织正式的迎新会，作为一个“day 0”活动，或者在午餐时间组织迎新会，让新成员认识其他社区成员，这样他们就知道以后可以向谁提问了。
```
